src/serac/physics/utilities/quadrature_data.hpp

```diff
@@ -200,49 +273,95 @@ class QuadratureData : public SyncableData {
  */
 template <>
 class QuadratureData<void> {
+  // Doesn't do anything
+public:
+  SERAC_HOST_DEVICE std::nullptr_t operator()(const int, const int) { return nullptr; }
 };
 
 // A dummy global so that lvalue references can be bound to something of type QData<void>
 // FIXME: There's probably a cleaner way to do this, it's technically a non-const global
 // but it's not really mutable because no operations are defined for it
 extern QuadratureData<void> dummy_qdata;
 
+/**
+ * @brief Stores instances of user-defined type for each quadrature point in a mesh
+ * @tparam T The type of the per-qpt data
+ * @pre T must be default-constructible (TODO: Do we want to allow non-default constructible types?)
+ * @pre T must be trivially copyable (due to the use of memcpy for type punning)
+ */
+template <typename T>
+class QuadratureDataView : public QuadratureDataImpl<T, QuadratureDataView<T>> {
+public:
+  QuadratureDataView(QuadratureData<T>& quad_data) : data_(quad_data.data()), offsets_(quad_data.offsets()) {}
+  SERAC_HOST_DEVICE T*         data() { return data_; }
+  SERAC_HOST_DEVICE const int* offsets() const { return offsets_; }
+
+private:
+  T*         data_    = nullptr;
+  const int* offsets_ = nullptr;
+};
+
+template <typename T>
+QuadratureDataView(QuadratureData<T>&) -> QuadratureDataView<T>;
```

Comment:
Is this user-defined CTAD guide needed? The implicitly generated one appears to work: https://godbolt.org/z/739fs3958